Gems/Multiplayer/Code/Source/NetworkEntity/NetworkEntityManager.cpp

```diff
@@ -467,8 +472,87 @@ namespace Multiplayer
         return netEntityId;
     }
 
-    void NetworkEntityManager::OnRootSpawnableAssigned(
-        [[maybe_unused]] AZ::Data::Asset<AzFramework::Spawnable> rootSpawnable, [[maybe_unused]] uint32_t generation)
+    AZStd::unique_ptr<AzFramework::EntitySpawnTicket> NetworkEntityManager::RequestNetSpawnableInstantiation(
+        const AZ::Data::Asset<AzFramework::Spawnable>& rootSpawnable, const AZStd::vector<AZ::Entity*>& entities)
+    {
+        if (entities.empty())
+        {
+            AZ_Error("NetworkEntityManager", false,
+                "RequestNetSpawnableInstantiation: No entities in the spawnable %s", rootSpawnable.GetHint().c_str());
+            return nullptr;
+        }
+
+        // The first entity in every spawnable is the root one 
+        const AZ::Entity* rootEntity = *entities.begin();
+        if(!rootEntity)
+        {
+            AZ_Error("NetworkEntityManager", false,
+                "RequestNetSpawnableInstantiation: Root entity is null in the spawnable %s", rootSpawnable.GetHint().c_str());
+            return nullptr;
+        }
+
+        const auto* holderComponent = rootEntity->FindComponent<NetworkSpawnableHolderComponent>();
+        if(!holderComponent)
+        {
+            // This spawnable doesn't have a corresponding network spawnable.
+            return nullptr;
+        }
+
+        AzFramework::TransformComponent* rootEntityTransform =
+            rootEntity->FindComponent<AzFramework::TransformComponent>();
+        if (!rootEntityTransform)
```

Comment:
Is this actually possible? I guess if the transform is removed in JSON?

Reply:
Probably impossible with the current tooling, but the data comes from the asset, so anything can happen.
Null check isn't a lot, but at least we won't crash here and just log an error with investigation info